core-rust/core-api-server/src/core_api/helpers.rs

```diff
@@ -45,6 +45,24 @@ pub(crate) fn read_known_substate(
     )
 }
 
+#[tracing::instrument(skip_all)]
+pub(crate) fn read_known_substate_from_id(
```

Comment:
(nitpick)

```suggestion
pub(crate) fn read_mandatory_substate_from_id(
```

or just `.read_substate_from_id()`? to have a clear distinction against the `read_optional_...()` family.

Reply:
How about eg `read_mandatory_substate_from_id_or_server_error`?

Reply:
whenever an `_or_` comes up in a method name, I can't help but recall a 2012 meme:
![image](https://user-images.githubusercontent.com/119413677/226756914-b7a72472-a89e-4960-b6f3-3a573e87bfc8.png)

(I mean: what else a method, which returns a `Result<>`, can do, when a mandatory substate turns out to be absent?)

Still, this is only about naming, so feel free to ignore my sentiment :)

Reply:
Hahaha love it. Well here `_or` really means `_else_` but yeah - happy to go back to `read_mandatory_substate_from_id` and not over correct! Will do that tomorrow